hpx/runtime/threads/executors/limiting_executor.hpp

```diff
@@ -44,34 +43,40 @@ namespace hpx { namespace threads { namespace executors
         //
         // --------------------------------------------------------------------
         limiting_executor(std::size_t lower, std::size_t upper,
-                          bool block_on_destruction=true)
-            : executor_(Executor())
-            , count_(0)
-            , lower_threshold_(lower)
-            , upper_threshold_(upper)
-            , block_(block_on_destruction) {}
-
-        limiting_executor(const Executor& ex, std::size_t lower, std::size_t upper,
-                            bool block_on_destruction=true)
-            : executor_(ex)
-            , count_(0)
-            , lower_threshold_(lower)
-            , upper_threshold_(upper)
-            , block_(block_on_destruction) {}
+            bool block_on_destruction = true)
+          : executor_(Executor())
+          , count_(0)
+          , lower_threshold_(lower)
+          , upper_threshold_(upper)
+          , block_(block_on_destruction)
+        {
+        }
+
+        limiting_executor(const Executor& ex, std::size_t lower,
+            std::size_t upper, bool block_on_destruction = true)
+          : executor_(ex)
+          , count_(0)
+          , lower_threshold_(lower)
+          , upper_threshold_(upper)
+          , block_(block_on_destruction)
+        {
+        }
 
         ~limiting_executor()
         {
-            if (block_) {
-                set_and_wait(0,0);
+            if (block_)
+            {
+                set_and_wait(0, 0);
             }
         }
 
         void count_up()
         {
-            if (++count_ > upper_threshold_) {
-                hpx::util::yield_while([&](){
-                    return (count_ > lower_threshold_);
-                });
+            if (++count_ > upper_threshold_)
+            {
+                hpx::util::yield_while(
+                    [&]() { return (count_ > lower_threshold_); }, nullptr,
+                    hpx::threads::pending, false);
```

Comment:
Unrelated to this PR: why isn't `limiting_executor` using a sliding semaphore (or similar) to actually suspend the 'waiting' threads instead of constantly rescheduling them?